.github/workflows/pipeline.yml

```diff
@@ -1,20 +1,38 @@
 name: Pipeline
 
 on:
-  push:
-    branches: [main]
   pull_request:
     # The branches below must be a subset of the branches above
     branches: [main, next]
 
 jobs:
-  CI:
-    name: CI
+  linux:
```

Comment:
Interesting to split them up by OS. Could we try adding windows back?

Reply:
Sure, though I don't see the value of that unless we start publishing Windows binaries, which we could certainly do.  But IIRC we had problems with the tests on Windows a long time ago. Also the Azure-specific behavior applies to workloads in Windows and Linux VMs so we'd have to run in Docker on Windows which would make things even slower

Reply:
Yeah, you're right, we are not doing windows binaries so no need